evcouplings/couplings/model.py

```diff
@@ -397,6 +387,30 @@ def __read_plmc_v2(self, filename, precision):
                     )
                     self.J_ij[j, i] = self.J_ij[i, j].T
 
+            # if lambda_h is negative, the model was
+            # inferred using mean-field
+            if self.lambda_h < 0:
+                # cast model to mean field model
+                from evcouplings.couplings.mean_field import MeanFieldCouplingsModel
+                self.__class__ = MeanFieldCouplingsModel
+
+                # pseudo-count is saved in lambda_h
```

Comment:
For complete perfection, we could move all the initialization below this line to its own function in MeanFieldCouplingsModel so all behaviour specific to mean-field is encapsulated there... but I know I am nitpicking ;)